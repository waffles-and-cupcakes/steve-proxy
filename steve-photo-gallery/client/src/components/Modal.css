.modal {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.75);
  z-index: 1001;
}

.modalContent {
  position: absolute;
  top: 10%;
  left: 15%;
  right: 15%;
  bottom: 20%;
  text-align: center;
  z-index: 1001;
}

.modalPhoto {
  height: 70vh;
}

.modalClose {
  background: rgba(0, 0, 0, 0);
  width: 100px;
  position: fixed;
  top: 0;
  right: 0;
  color: white;
  cursor: pointer;
  padding: 16px 28px 8px;
  border: none;
  outline: none;
}

.modalClose:active,
.modalClose:focus {
  background: rgba(0, 0, 0, 0);
}

.closeX {
  stroke: white;
  fill: transparent;
  stroke-linecap: round;
  stroke-width: 1;
}

svg {
  display: block;
}