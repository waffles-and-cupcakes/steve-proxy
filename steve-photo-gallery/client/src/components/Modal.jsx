import React from 'react';
import Carousel from './Carousel';
import style from './Modal.css';

const Modal = (props) => {
  return props.open ? (
    <div>
      <div className={style.modal}>
        <button className={style.modalClose} onClick={props.toggleModal}>
          <svg viewBox="0 0 40 40">
            <path className={style.closeX} d="M 10,10 L 30,30 M 30,10 L 10,30" />
          </svg>
        </button>
        <div className={style.modalContent}>
          <Carousel photos={props.photos} />
        </div>
      </div>
    </div>
  ) : null;
};

export default Modal;