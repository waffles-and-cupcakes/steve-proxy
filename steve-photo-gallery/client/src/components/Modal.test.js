import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Modal from './Modal.jsx';

configure({adapter: new Adapter() });

test('Gallery should render correctly', () => {
  const output = shallow(<Modal />);
  expect(shallowToJson(output)).toMatchSnapshot();
});