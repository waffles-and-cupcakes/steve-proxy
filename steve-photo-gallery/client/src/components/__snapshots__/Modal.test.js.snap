// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render correctly 1`] = `""`;
